node/src/components/linear_chain/pending_signatures.rs

```diff
@@ -64,8 +94,10 @@ impl PendingSignatures {
             );
             return false;
         }
+
         // Add the pending signature.
-        sigs.insert(block_hash, signature);
+        let value = VerificationStatus::Unknown(signature);
+        sigs.insert(block_hash, value);
```

Comment:
Let's make sure we don't overwrite a `Bonded` one with `Unknown` if we receive it twice.

Reply:
We check if the signature is already _pending_ before adding it to this collection.